### PR TITLE
feat: add HTTP API list to reference

### PR DESCRIPTION
**Pull request by waynexia**
## What's Changed in this PR

List all available HTTP endpoints and their usages.

## Checklist

- [ ] Please confirm that all corresponding versions of the documents have been revised.
- [ ] Please ensure that the content in `sidebars.ts` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/reference/http-endpoints.md [hunk 1]
```diff
@@ -0,0 +1,190 @@
+# HTTP API endpoint list
+
+Here is the full list for the various HTTP paths and their usage in GreptimeDB:
+
+## Miscellaneous
+
+Endpoints that is not versioned (under `/v1`). For miscellaneous usage like health check, status, metrics, etc.
```
Comment (waynexia) [inline docs/reference/http-endpoints.md:7]:
```suggestion
Endpoints that is not versioned (under `/v1`). For admin usage like health check, status, metrics, etc.
```

File: docs/reference/http-endpoints.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+- **Paths**:
+  - `/v1/prometheus/write`
+  - `/v1/prometheus/read`
+- **Methods**: `POST`
+- **Description**: Supports Prometheus remote write and read APIs.
+- **Usage**:
+  - Send metric data using Prometheus remote write protocol.
+  - Read metric data using Prometheus remote read protocol.
+
+### Prometheus HTTP API
```
[the remark this answers is not in the document]
Comment (waynexia) [reply docs/reference/http-endpoints.md:118]: This is not only for query. This group of endpoints is aimed to provide prometheus-compatible HTTP API as documented here https://prometheus.io/docs/prometheus/latest/querying/api/

File: docs/reference/http-endpoints.md [hunk 3]
```diff
@@ -0,0 +1,190 @@
+# HTTP API endpoint list
```
Comment (nicecui) [inline docs/reference/http-endpoints.md:1]:
```suggestion
# HTTP API Endpoint List
```

File: docs/reference/http-endpoints.md [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+- **Usage**: Access these endpoints to interact with the web-based dashboard.
+
+This dashboard is packaged with the GreptimeDB server and provides a user-friendly interface for interacting with the server. It requires corresponding compile flags to be enabled when building GreptimeDB. The original source code for the dashboard can be found at https://github.com/GreptimeTeam/dashboard
+
+### Debugging Tools
+
+- **Base Path**: `/debug`
+- **Endpoints**:
+  - `/log_level`
+  - `/pprof/*`
```
Comment (nicecui) [inline docs/reference/http-endpoints.md:51]: What is `*` in `/pprof/*`? Please write in detail.

File: docs/reference/http-endpoints.md [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+
+Various query APIs for sending query to GreptimeDB.
+
+### SQL API
+
+- **Path**: `/v1/sql`
+- **Methods**: `GET`, `POST`
+- **Description**: Executes SQL queries against the server.
+- **Usage**:
+  - `GET`: Send SQL queries via query parameters.
```
Comment (nicecui) [inline docs/reference/http-endpoints.md:70]:
```suggestion
```
As discussed before, We do not recommend use the GET method to query data. All SQL API should use `POST`, which means post SQL statements according to the RESTful Principles.

File: docs/reference/http-endpoints.md [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+  - `POST`: Send SQL queries in the request body.
+  
+For more information on the SQL API, refer to the [SQL documentation](/user-guide/query-data/sql.md).
```
Comment (nicecui) [inline docs/reference/http-endpoints.md:73]:
```suggestion
For more information on the SQL API, refer to the [HTTP API documentation](/user-guide/protocols/http.md#post-sql-statements) in the user guide.
```

**Review by fengjiachun — COMMENTED** [review]
LGTM, please add the CH doc

File: i18n/zh/docusaurus-plugin-content-docs/current/reference/http-endpoints.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+## 查询端点
+
+用于向 GreptimeDB 发送查询的各种查询 API。
+
+### SQL API
+
+- **路径**: `/v1/sql`
+- **方法**: `GET`, `POST`
+- **描述**: 对服务器执行 SQL 查询。
```
Comment (fengjiachun) [inline i18n/zh/docusaurus-plugin-content-docs/current/reference/http-endpoints.md:75]:
```suggestion
- **描述**: 执行 SQL 查询。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/reference/http-endpoints.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+- **用法**: 在请求体中发送 SQL 查询。
```
Comment (fengjiachun) [inline i18n/zh/docusaurus-plugin-content-docs/current/reference/http-endpoints.md:76]:
```suggestion
- **用法**: 在请求体中发送 SQL 语句。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/reference/http-endpoints.md [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+
+有关 SQL API 的更多信息，请参阅用户指南中的 [HTTP API 文档](/user-guide/protocols/http.md#post-sql-statements)。
+
+### PromQL API
+
+- **路径**: `/v1/promql`
+- **方法**: `GET`, `POST`
+- **描述**: 执行 PromQL 查询以获取 Prometheus 兼容的指标，并以 GreptimeDB 的 JSON 格式返回数据。
+- **用法**: 在请求体中发送 PromQL 查询。
```
Comment (fengjiachun) [inline i18n/zh/docusaurus-plugin-content-docs/current/reference/http-endpoints.md:85]:
```suggestion
- **用法**: 在请求体中发送 PromQL 语句。
```

**Review by fengjiachun — APPROVED** [review]
Rest LGTM

**Review by nicecui — APPROVED** [review]

**Review by nicecui — COMMENTED** [review]
please also modify the zh HTTP protocol document in user guide.

**Pull request merged**